### DISCUSSION TITLE
How to finish with result and examine result in a parent

### DISCUSSION THREAD

**Discussion by dimsuz** [bumble-tech/appyx#279] [General]
What is the idiomatic way to finish with result? For example:

* `AppFlow` starts `LoginFlow`
* `LoginFlow` can end with "Dismissed", "Success", "AdditionalIdentificationRequired" statuses
* `AppFlow` should be able to somehow receive these outcomes and decide where to route next

If I'd express this with lambdas it could be something like:

```kotlin
appFlow {
   val result = loginFlow()
   when (result) {
     is Dismissed -> Unit
     is Success -> mainFlow()
     is AdditionalIdentificationRequired -> identifyFlow(result.idLevel)
}
```

**Comment by KovalevAndrey** [comment]
Hi @dimsuz, thanks for your question. 

You should use [childAware API](https://bumble-tech.github.io/appyx/apps/childaware/#childaware-api) to establish connection between parent and a child. Then it's up to you how you communicate between the parent and the child. 

You could use simple callbacks or provided by library reactive solutions `com.bumble.appyx.interop.rx2.connectable.NodeConnector`.

**Reply by dimsuz** [reply]
Thanks! Is there analog of NodeConnector for kotlin's coroutines `Flow`? Didn't find it in libraries folder. Not sure I need it, and perhaps it would be hard to write it, just curious if I missed something.

**Reply by KovalevAndrey** [reply]
Unfortunately, at the moment we don't have an analogue for coroutines

**Reply by RationalRank** [reply]
I've [created](https://gist.github.com/RationalRank/c7e493ac04aa0567c6f1cd65f709335b) a coroutines variant of Appyx's `NodeConnector`. Open to feedbacks :)

**Reply by zsoltk** [reply]
Hey @RationalRank we have #284. A PR is very welcome!
